app/scripts/modules/core/pipeline/config/stages/executionWindows/executionWindowsDetails.controller.js

```diff
@@ -13,9 +15,28 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.executionWindows.
 
     $scope.configSections = ['windowConfig', 'taskStatus'];
 
+    // yes, this is ugly - when we replace the execution window w/ an ng2 component, this will go away.  i promise
+    function replaceDays(days) {
+      const daySet = new Set(days);
+      return DAYS_OF_WEEK.filter(day => daySet.has(day.ordinal)).map(day => day.label);
+    }
+
+    // ditto
+    function getDayText() {
+      let dayText = 'Everyday';
+      const days = $scope.stage.context.restrictedExecutionWindow.days;
+      if (days && (days.length > 0)) {
+        const daysAsText = replaceDays(days);
+        dayText = daysAsText.toString().replace(/,/g, ', ');
```

Comment:
If you weren't using a `Set` above, you could just do `days.join(', ')` here. But this works, too, and I know how much you really like your `Set`s.